src/validate-npm-package/validate-npm-package.ts

```diff
@@ -0,0 +1,95 @@
+// validate-npm-package/validate-npm-package.ts
+
+import os from 'node:os';
+import path from 'node:path';
+import { promises as fs } from 'node:fs';
+import childProcess from 'node:child_process';
+import util from 'node:util';
+
+import debug from 'debug';
+import { v4 as uuid } from 'uuid';
+
+import { addNPMRCFile } from '../publish-beta/publish';
+
+interface PackageJson {
+  type?: string;
+  service?: {
+    api: {
+      root: string;
+      endpoints: string[];
+    };
+  };
+}
+
+const exec = util.promisify(childProcess.exec);
+const log = debug('github-actions:validate-npm-package');
+
+async function obtainNpmPackage(packageVersion: string, workFolder: string): Promise<void> {
+  await addNPMRCFile(workFolder);
+
+  const fullCommandLine = `npm pack ${packageVersion}`;
```

Comment:
could we do an `npm install` somewhere instead?  then we could do an `import()` and make sure all the machinery works (including the package.json is ok)